frontend/src/components/pages/knowledgebase/KnowledgeBase.List.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  createStandaloneToast,
+  DataTable,
+  Flex,
+  HStack,
+  Icon,
+  Image,
+  SearchField,
+  Stack,
+  Text,
+} from '@redpanda-data/ui';
+import { makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import { AiOutlineDelete } from 'react-icons/ai';
+import { Link } from 'react-router-dom';
+import EmptyConnectors from '../../../assets/redpanda/EmptyConnectors.svg';
+import type { KnowledgeBase } from '../../../protogen/redpanda/api/dataplane/v1alpha1/knowledge_base_pb';
+import { appGlobal } from '../../../state/appGlobal';
+import { knowledgebaseApi } from '../../../state/backendApi';
+import { Features } from '../../../state/supportedFeatures';
+import { uiSettings } from '../../../state/ui';
+import { openModal } from '../../../utils/ModalContainer';
+import { DefaultSkeleton } from '../../../utils/tsxUtils';
+import { encodeURIComponentPercents } from '../../../utils/utils';
+import PageContent from '../../misc/PageContent';
+import { PageComponent, type PageInitHelper } from '../Page';
+import { ExplicitConfirmModal } from '../rp-connect/modals';
+
+const { ToastContainer, toast } = createStandaloneToast();
+
+function openDeleteKnowledgeBaseModal(knowledgeBaseName: string, onConfirm: () => void) {
+  openModal(ExplicitConfirmModal, {
+    title: <>Permanently delete knowledge base {knowledgeBaseName}</>,
+    body: <>Deleting a knowledge base cannot be undone.</>,
+    primaryButtonContent: <>Delete</>,
+    secondaryButtonContent: <>Cancel</>,
+
+    requiredText: knowledgeBaseName.trim(),
+
+    onPrimaryButton: (closeModal) => {
+      onConfirm();
+      closeModal();
+    },
+
+    onSecondaryButton: (closeModal) => {
+      closeModal();
+    },
+  });
+}
+
+const CreateKnowledgeBaseButton = () => {
+  return (
+    <ButtonGroup>
+      <Button variant="outline" data-testid="create-knowledge-base-button">
+        <Link to={'/knowledgebases/create'} style={{ textDecoration: 'none', color: 'inherit' }}>
+          Create knowledge base
+        </Link>
+      </Button>
+    </ButtonGroup>
+  );
+};
+
+const EmptyPlaceholder = () => {
+  return (
+    <Flex alignItems="center" justifyContent="center" flexDirection="column" gap="4" mb="4">
+      <Image src={EmptyConnectors} />
+      <Box>You have no knowledge bases.</Box>
+      <CreateKnowledgeBaseButton />
+    </Flex>
+  );
+};
+
+@observer
+class KnowledgeBaseList extends PageComponent<{}> {
```

Comment:
This is an outdated pattern as well.

We should use functional React components (not class-based), and try to do something similar to frontend/src/components/pages/agents/create/create-agent-page.tsx where you do not rely on any MobX state at all.

Reply:
done, could you check the result is what you expected?